core/src/com/ericc/the/game/systems/logic/ActionHandlingSystem.java

```diff
@@ -0,0 +1,174 @@
+package com.ericc.the.game.systems.logic;
+
+import com.badlogic.ashley.core.Engine;
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.ashley.core.EntitySystem;
+import com.badlogic.ashley.core.Family;
+import com.badlogic.ashley.utils.ImmutableArray;
+import com.badlogic.gdx.math.GridPoint2;
+import com.ericc.the.game.Direction;
+import com.ericc.the.game.Mappers;
+import com.ericc.the.game.actions.Actions;
+import com.ericc.the.game.actions.MovementAction;
+import com.ericc.the.game.components.*;
+import com.ericc.the.game.entities.Mob;
+import com.ericc.the.game.entities.Player;
+import com.ericc.the.game.entities.PushableObject;
+import com.ericc.the.game.map.Map;
+
+import java.util.AbstractMap;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class ActionHandlingSystem extends EntitySystem {
+
+    private ImmutableArray<Entity> movables;
+    private ArrayList<AbstractMap.SimpleEntry<Integer, Entity>> movableInitiatives;
+    private HashMap<GridPoint2, Entity> interactives;
+    private Map map;
+
+    public ActionHandlingSystem(Map map) {
+        super(10001);
+        this.map = map;
+    }
+
+    // Comparator for the AbstractMap.SimpleEntry<> class
+    // The List will be sorted in descending order
+    public class pairComparator implements Comparator<AbstractMap.SimpleEntry<Integer, Entity>> {
+        @Override
+        public int compare(AbstractMap.SimpleEntry<Integer, Entity> integerEntitySimpleEntry,
+                           AbstractMap.SimpleEntry<Integer, Entity> t1) {
+            return Integer.compare(t1.getKey(), integerEntitySimpleEntry.getKey());
+        }
+    }
+
+    @Override
+    public void addedToEngine(Engine engine) {
+        movables = engine.getEntitiesFor(Family.all(DirectionComponent.class,
+                PositionComponent.class, CurrentActionComponent.class,
+                StatisticsComponent.class, SentienceComponent.class).get());
+        movableInitiatives = new ArrayList<>();
+        interactives = new HashMap<>();
+
+        ImmutableArray<Entity> temp;
+
+        temp = engine.getEntitiesFor(Family.all(PositionComponent.class,
+                InteractivityComponent.class, CurrentActionComponent.class).get());
+
+        for (Entity entity : temp) {
+            PositionComponent pos = Mappers.position.get(entity);
+            interactives.put(new GridPoint2(pos.x, pos.y), entity);
+        }
+    }
+
+    @Override
+    public void update(float deltaTime) {
+        // Rolling for initiative for every entity that is capable of independent decision-making
+        for (Entity entity : movables) {
```

Comment:
To jak nic zasługuje na osobny system, niech dodaje się tam komponent Initiative